app/assets/javascripts/admin/views/team/team_list_view.js

```diff
@@ -1,63 +1,156 @@
+
+const { Column } = Table;
+const { Search } = Input;
+
+type State = {
+  isLoading: boolean,
+  teams: Array<APITeamType>,
+  searchQuery: string,
+  isTeamCreationModalVisible: boolean,
+};
 
-    this.prototype.ui = { modalWrapper: ".modal-wrapper" };
+class TeamListView extends React.PureComponent<{}, State> {
+  state = {
+    isLoading: true,
+    teams: [],
+    searchQuery: "",
+    isTeamCreationModalVisible: false,
+  };
+
+  componentDidMount() {
+    this.fetchData();
   }
 
-  initialize() {
-    this.listenTo(app.vent, "paginationView:filter", this.filterBySearch);
-    this.listenTo(app.vent, "modal:destroy", this.render);
-    this.listenTo(app.vent, "paginationView:addElement", this.showModal);
+  async fetchData(): Promise<void> {
+    const url = "/api/teams?isEditable=true";
+    const teams = await Request.receiveJSON(url);
 
-    return this.collection.fetch({
-      data: "isEditable=true",
+    this.setState({
+      isLoading: false,
+      teams: teams.filter(p => p.owner),
     });
   }
 
-  filterBySearch(filterQuery) {
-    return this.collection.setFilter(["name", "owner"], filterQuery);
-  }
+  handleSearch = (event: SyntheticInputEvent<>): void => {
+    this.setState({ searchQuery: event.target.value });
+  };
+
+  deleteTeam = async (script: APITeamType) => {
+    Modal.confirm({
+      title: messages["team.delete"],
+      onOk: () => {
+        this.setState({
+          isLoading: true,
+        });
+
+        const url = `/api/teams/${script.id}`;
+        Request.receiveJSON(url, {
+          method: "DELETE",
+        }).then(() => {
+          this.setState({
+            isLoading: false,
+            teams: this.state.teams.filter(p => p.id !== script.id),
+          });
+        });
+      },
+    });
+  };
+
+  createTeam = async (newTeam: APITeamType) => {
+    this.setState({
+      isTeamCreationModalVisible: false,
+      teams: this.state.teams.concat([newTeam]),
+    });
+  };
+
+  render() {
+    const marginRight = { marginRight: 20 };
 
-  showModal(modalView) {
-    modalView = new CreateTeamModalView({ teamCollection: this.collection });
-    this.ui.modalWrapper.html(modalView.render().el);
+    return (
+      <div className="container wide">
+        <div style={{ marginTag: 20 }}>
+          <div className="pull-right">
+            <Button
+              icon="plus"
+              style={marginRight}
+              type="primary"
+              onClick={() => this.setState({ isTeamCreationModalVisible: true })}
+            >
+              Add Team
+            </Button>
+            <Search
+              style={{ width: 200 }}
+              onPressEnter={this.handleSearch}
+              onChange={this.handleSearch}
+            />
+          </div>
+          <h3>Teams</h3>
+          <div className="clearfix" style={{ margin: "20px 0px" }} />
 
-    return modalView.show();
+          <Spin spinning={this.state.isLoading} size="large">
```

Comment:
Cool, I didn't know, that you could wrap the loaded content within a Spin.

Reply:
The good part is that the spinner will center around the wrapped content.